test/misc/migrations_test.rb

```diff
@@ -0,0 +1,36 @@
+require "test_helper"
+
+class MigrationsTest < ActiveSupport::TestCase
+  test "don't modify tables in production" do
+    modification_calls = %w[
+      create_table
+      change_table
+      drop_table
+      rename_table
+
+      add_index
+      rename_index
+      remove_index
+
+      add_foreign_key
+      remove_foreign_key
+
+      add_column
+      rename_column
+      remove_column
+      remove_columns
+      change_column
+      change_column_null
+    ]
+
+    ApplicationRecord.connection.migration_context.migrations.each do |migration_proxy|
+      # We only started running migrations manually after the below time
+      next if migration_proxy.version <= 20231017101049 # rubocop:disable Style/NumericLiterals
+
+      migration = File.read(migration_proxy.filename)
+      next unless modification_calls.any? { |modification_call| migration.include?(modification_call) }
```

Comment:
I sort of feel like we should't run *any* code automtically in production, so maybe removing this and just hasving the line below is actually fine.